### PR TITLE
Feat/backend

### PR DESCRIPTION
**Pull request by Lem0nTree**
- Added backend Routes
- Added database schema
- Added openAI api to trained model
- Added NEAR conversion to USD , cached 12 hours

### CODE REVIEW

File: app/.claudeignore
Comment (urataps) [inline app/.claudeignore:1]: Bro why do we need .claudeignore files? Is this built-in feature of cursor?

File: app/.claudesync/config.local.json
Comment (urataps) [inline app/.claudesync/config.local.json:1]: What this file does sir?

File: app/src/app/layout.tsx [hunk 1]
```diff
@@ -19,15 +20,17 @@ export default function RootLayout({
...
+          <div className="min-h-screen w-64 bg-foreground">
+            <SidebarMenu />
+          </div>
+          <main className="flex-1 overflow-y-auto bg-background p-6">
+            <TRPCReactProvider>
+              <Navbar />
+              {children}
+            </TRPCReactProvider>
+          </main>
+        </ToastProvider>
```
Comment (urataps) [inline app/src/app/layout.tsx:33]: ToastProvider does not have to be a wrapper can be a simple <ToastProvider/> at the end

File: app/src/server/api/routers/strategy.ts [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+      try {
+        // Ensure user exists in database
+        const user = await ctx.db.user.upsert({
+          where: { id: input.userId },
+          create: { id: input.userId },
+          update: {},
+        });
+
+        // Convert NEAR amount to number
+        const nearAmount = parseFloat(input.neat);
```
Comment (urataps) [inline app/src/server/api/routers/strategy.ts:50]: input.neat?

File: app/src/server/api/routers/strategy.ts
Comment (urataps) [inline app/src/server/api/routers/strategy.ts:1]: I'm proud of you sir, this is top notch big brain backend (no joke)

**Review by urataps — APPROVED** [review]

File: app/.claudeignore
Comment (Lem0nTree) [reply app/.claudeignore:1]: Kind of , is an extension i use in cursor to sync claude projects

File: app/.claudesync/config.local.json
Comment (Lem0nTree) [reply app/.claudesync/config.local.json:1]: can add it to ignore file

File: app/src/app/layout.tsx (hunk 1 above)
Comment (Lem0nTree) [reply app/src/app/layout.tsx:33]: I tried without wrapping but is giving me error

File: app/src/server/api/routers/strategy.ts (hunk 2 above)
Comment (Lem0nTree) [reply app/src/server/api/routers/strategy.ts]: fixed typo

**Comment by Lem0nTree** [comment]
- Added skeleton while loading
- Added dynamic icons for chains and protocol
- Added dynamic data in the investment card
- Added tooltip on hover
- Added custom chain colors
- Improved protocols icons

**Review by urataps — APPROVED** [review]

**Pull request merged**